pages/nfts/new.tsx

```diff
@@ -0,0 +1,323 @@
+import { Form, Layout } from 'antd';
+import React, { useContext, useEffect, useReducer } from 'react';
+import styled from 'styled-components';
+import StepWizard from 'react-step-wizard';
+import Upload from '@/modules/nfts/components/wizard/Upload';
+import Verify from '@/modules/nfts/components/wizard/Verify';
+import InfoScreen from '@/modules/nfts/components/wizard/InfoScreen';
+import { useForm } from 'antd/lib/form/Form';
+import Summary from '@/modules/nfts/components/wizard/Summary';
+import RoyaltiesCreators from '@/modules/nfts/components/wizard/RoyaltiesCreators';
+import { WalletContext } from '@/modules/wallet';
+import PriceSummary from '@/modules/nfts/components/wizard/PriceSummary';
+import MintInProgress from '@/modules/nfts/components/wizard/MintInProgress';
+import { isNil } from 'ramda';
+import OffRampScreen from '@/modules/nfts/components/wizard/OffRamp';
+import { Connection } from '@solana/web3.js';
+
+export const MAX_CREATOR_LIMIT = 5;
+
+export interface Creator {
+  address: string;
+  share: number;
+}
+
+const StyledLayout = styled(Layout)`
+  width: 100%;
+  overflow: hidden;
+`;
+
+export interface UploadedFilePin {
+  name: string;
+  uri: string;
+  type: string;
+}
+
+export type FormValues = { [key: string]: NFTFormValue };
+
+export interface NFTFormValue {
+  name: string;
+  imageName: string;
+  description: string;
+  collection: string;
+  attributes: Array<NFTAttribute>;
+  seller_fee_basis_points: number;
+  properties: { creators: Array<Creator>; maxSupply: number };
+}
+
+export type FileOrString = UploadedFilePin | string;
+
+export type NFTAttribute = {
+  trait_type: string;
+  value: string | number;
+};
+
+export enum MintStatus {
+  FAILED,
+  SUCCESS,
+}
+
+export interface NFTValue {
+  name: string;
+  description: string;
+  attributes?: NFTAttribute[];
+  collection?: string;
+  seller_fee_basis_points: number;
+  mintStatus?: MintStatus;
+
+  properties: {
+    files?: FileOrString[];
+    maxSupply: number;
+    creators?: Creator[];
+  };
+}
+
+export type MintDispatch = (action: MintAction) => void;
+
+interface State {
+  images: Array<File>;
+  uploadedFiles: Array<UploadedFilePin>;
+  formValues: NFTFormValue[] | null;
+  nftValues: NFTValue[];
+}
+
+export interface MintAction {
+  type:
+    | 'SET_IMAGES'
+    | 'DELETE_IMAGE'
+    | 'ADD_IMAGE'
+    | 'UPLOAD_FILES'
+    | 'SET_FORM_VALUES'
+    | 'SET_NFT_VALUES';
+  payload: File[] | File | String | Array<UploadedFilePin> | NFTFormValue[] | NFTValue[];
+}
+
+const initialState: State = {
+  images: [],
+  uploadedFiles: [],
+  formValues: null,
+  nftValues: [],
+};
```

Comment:
Perhaps make this a function returning an initial state?  This would help in mitigating bugs with erroneously effectful code.
```suggestion
const initialState = (): State => {
  return {
    images: [],
    uploadedFiles: [],
    formValues: null,
    nftValues: [],
  };
};
```